dftimewolf/lib/processors/gcp_cloud_resource_tree_helper.py

```diff
@@ -0,0 +1,303 @@
+      elif resource_type == 'instances':
+        self.type = 'gce_instance'
+      elif resource_type == 'images':
+        self.type = 'gce_image'
+      elif resource_type == 'machineImages':
+        self.type = 'gce_machine_image'
+      elif resource_type == 'instanceTemplates':
+        self.type = 'gce_instance_template'
+      elif resource_type == 'snapshots':
+        self.type = 'gce_snapshot'
+      else:
+        self.type = resource_type
+
+      if '/zones/' in value or '/regions/' in value:
+        self.zone = values[-3]
+        self.project_id = values[-5]
+      elif '/global/' in value:
+        self.zone = 'global'
+        self.project_id = values[-4]
+
+    self._resource_name = value
+
+  @property
+  def creation_timestamp(self) -> Optional[datetime]:
+    """Property creation_timestamp Getter."""
+    return self._creation_timestamp
+
+  @creation_timestamp.setter
+  def creation_timestamp(self, value: Union[datetime, str]) -> None:
+    """Property creation_timestamp Setter.
+
+    Args:
+      value: value to set the creation_timestamp to.
+    """
+    if isinstance(value, datetime):
+      self._creation_timestamp = value
+    else:
+      self._creation_timestamp = parser.parse(value)
+
+  @property
+  def deletion_timestamp(self) -> Optional[datetime]:
+    """Property deletion_timestamp Getter."""
+    return self._deletion_timestamp
+
+  @deletion_timestamp.setter
+  def deletion_timestamp(self, value: Union[datetime, str]) -> None:
+    """Property deletion_timestamp Setter.
+
+    Args:
+      value: value to set the deletion_timestamp to.
+    """
+    if isinstance(value, datetime):
+      self._deletion_timestamp = value
+    else:
+      self._deletion_timestamp = parser.parse(value)
+
+  def IsDeleted(self) -> bool:
+    """Checks if resource is deleted."""
+    if self.deleted or self.deletion_timestamp or not self.creation_timestamp:
+      return True
+
+    return False
+
+  def GenerateTree(self) -> List[Dict[str, 'Resource']]:
+    """Generates the resource tree.
+
+    Returns:
+      List of dictionaries containing a reference to the resource and its name
+      indented based on it's location in the tree.
+    """
+    tab = '\t'
+    output: List[Any] = []
+
+    # Count the number of parent resources and the level in the tree the
+    # resource is at.
+    level = 0
+    parent_resource = self.parent
+    while parent_resource:
+      level = level + 1
+      parent_resource = parent_resource.parent
+
+    # Add the resource parent entries to the List of dictionaries
+    counter = level
+    parent_resource = self.parent
+    if parent_resource:
+      while counter > 0:
+        counter = counter - 1
+        entry: Dict[str, Any] = {}
+        entry['resource_object'] = parent_resource
+        entry['graph'] = f'{tab*counter}|--{parent_resource.name}'
+        output.insert(0, entry)
+        if parent_resource.parent:
+          parent_resource = parent_resource.parent
```

Comment:
you start here from self.parent, create entry and then insert (0,entry), so you insert the entries in the following order: parent, grandparent, grandgrandparent, ...

then after the loop finishes you create entry for self and push it to the end of output with insert(level,entry). 

Why don't you move adding the resource entry before adding parents and insert it to (0,entry)?

this way the insertion happens in order and less confusing for reader

Reply:
I already replied to the same comment here https://github.com/log2timeline/dftimewolf/pull/540#discussion_r917949500 

TL;DR, doesn't make a difference. I add things in the same order they are displayed. Parent resources > Resource in question > Child Resources. Feels more readable to me. Also I need to determine the 'level' the resource is at which I use to determine the indentation. I don't know how many parents are there before hand.